R/helper_clean_model_data.R

```diff
@@ -139,7 +155,10 @@ clean_model_data <- function(data, datargs, estimator = "") {
 
   ret[["block"]] <- model.extract(mf, "block")
 
-  ret[["condition_pr"]] <- model.extract(mf, "condition_pr")
+  ret[["condition_pr"]] <- if (recycle_prs)
+    rep(condition_pr, nrow(ret[["design_matrix"]]))
```

Comment:
use rep_len, then you don't need the `recycle_prs` flag at all.

Reply:
They come from different sources, so I think I still need it.